cdk/lib/elastic-search-monitor.ts

```diff
@@ -1,15 +1,74 @@
-import { join } from "path";
-import { CfnInclude } from "@aws-cdk/cloudformation-include";
+import { SecurityGroup } from "@aws-cdk/aws-ec2";
+import { Schedule } from "@aws-cdk/aws-events";
+import { PolicyStatement } from "@aws-cdk/aws-iam";
+import { Runtime } from "@aws-cdk/aws-lambda";
 import type { App } from "@aws-cdk/core";
+import { CfnParameter, Duration } from "@aws-cdk/core";
+import { GuScheduledLambda } from "@guardian/cdk";
 import type { GuStackProps } from "@guardian/cdk/lib/constructs/core";
 import { GuStack } from "@guardian/cdk/lib/constructs/core";
+import { GuVpc } from "@guardian/cdk/lib/constructs/ec2";
 
 export class ElasticSearchMonitor extends GuStack {
   constructor(scope: App, id: string, props: GuStackProps) {
     super(scope, id, props);
-    const yamlTemplateFilePath = join(__dirname, "../..", "cfn.yaml");
-    new CfnInclude(this, "YamlTemplate", {
-      templateFile: yamlTemplateFilePath,
+    const clusterSecurityGroupId = new CfnParameter(
+      this,
+      "ClusterSecurityGroup",
+      {
+        type: "AWS::EC2::SecurityGroup::Id",
+        description:
+          "A security group allowing the lambda to connect to Elasticsearch on port 9200 over TCP",
+      }
+    );
+    const networkInterfacePolicy = new PolicyStatement({
+      actions: [
+        "ec2:CreateNetworkInterface",
+        "ec2:DescribeNetworkInterfaces",
+        "ec2:DeleteNetworkInterface",
+      ],
+      resources: ["*"],
     });
+    const ec2DetectionPolicy = new PolicyStatement({
+      actions: ["ec2:DescribeInstances"],
+      resources: ["*"],
+    });
+    const pushMetricsPolicy = new PolicyStatement({
+      actions: ["cloudwatch:PutMetricData"],
+      resources: ["*"],
+    });
+    const additionalPolicies: PolicyStatement[] = [
+      networkInterfacePolicy,
+      ec2DetectionPolicy,
+      pushMetricsPolicy,
+    ];
+    const scheduledLambda = new GuScheduledLambda(this, "ScheduledLambda", {
+      app: "elastic-search-monitor",
+      fileName: "elastic-search-monitor.jar",
+      environment: {
+        TAG_QUERY_APP: "elk-es-master",
+        TAG_QUERY_STACK: "deploy",
+        CLUSTER_NAME: "elk",
+      },
+      description:
+        "Monitors your Elasticsearch cluster and reports metrics to CloudWatch",
+      handler: "com.gu.elasticsearchmonitor.Lambda::handler",
+      monitoringConfiguration: { noMonitoring: true },
```

Comment:
Is it worth turning monitoring on (perhaps in a later PR?) or is there some reason it is not useful here?

Reply:
We should probably turn it on in case we ever break the lambda - will open a follow-up PR for that.

Reply:
I should have also mentioned that we currently define some alarms [in a separate template](https://github.com/guardian/elastic-search-monitor/blob/e0d3703bb5819f12ac9f398c97c3a738eb3f3a68/alarms.yaml). Based on the [`riff-raff.yaml`](https://github.com/guardian/elastic-search-monitor/blob/e0d3703bb5819f12ac9f398c97c3a738eb3f3a68/riff-raff.yaml) I think these must be deployed manually, so I will try to migrate those across too.

Reply:
https://github.com/guardian/elastic-search-monitor/pull/18